src/main/java/com/avispl/symphony/dal/communicator/crestron/CrestronXiO.java

```diff
@@ -323,10 +322,14 @@ protected void internalDestroy() {
 
     /**
      * {@inheritDoc}
+     * The timestamp for collected devices has to be updated every monitoring cycle since large amount of
+     * devices takes longer to go through (consider looping/requesting stats), so the device info is not
+     * considered stale. The device info list is still relevant.
      */
     @Override
     public List<AggregatedDevice> retrieveMultipleStatistics() throws Exception {
         updateValidRetrieveStatisticsTimestamp();
```

Comment:
The retrieveMultipleStatistics(deviceIds) is also making call to updateValidRetrieveStatisticsTimestamp()...Since that method already calls this, there is no point in calling updateValidRetrieveStatisticsTimestamp() within retrieveMultipleStatistics(deviceIds)